dags/general/monitoring.py

```diff
@@ -0,0 +1,133 @@
+from airflow import DAG
+import json
+from operators.mattermost_operator import MattermostOperator
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.http_operator import SimpleHttpOperator
+from datetime import datetime
+from airflow_utils import send_alert
+from airflow.models.xcom import XCom
+from airflow.utils.db import provide_session
+from airflow.models import Variable
+import logging
+task_logger = logging.getLogger('airflow.task')
+
+
+"""
+This method returns list of extractions that are failing
+tap_exit_status = 1, failing
+tap_exit_status = 0, success
+tap_exit_status = None, extraction in process
+"""
+def stitch_check_extractions(response):
+    task_logger.info('Got extractions response from stitch api, checking for errors')
+    failed_extractions={}
+    try:
+        extractions = json.loads(response)['data']
+        for extraction in extractions:
+            if extraction['tap_exit_status'] is 1:
+                failed_extractions[extraction['source_id']] = extraction['tap_description']
+    except Exception as e:
+        task_logger.error('error in getting extractions from stitch ', str(e))
```

Comment:
A better approach would be to use `task_logger.exception` or `task_logger.error("error in ...", exc_info=True)`. This method automatically adds exception information to the log message. See [logging documentation](https://docs.python.org/3/library/logging.html#logging.Logger.exception).